internal/tiltfile/k8s.go

```diff
@@ -66,6 +66,8 @@ type k8sResourceOptions struct {
 	triggerMode       triggerMode
 	tiltfilePosition  syntax.Position
 	resourceDeps      []string
+	// TODO(dmiller): this should be a different type once it has been parsed as a valid identifier
```

Comment:
docstring for this field pls

Reply:
Hmm none of the other fields are documented, unless you count https://docs.tilt.dev/api.html#api.k8s_resource as documentation. Should this one be an exception?